src/Illuminate/Cache/FileStore.php

```diff
@@ -267,9 +267,15 @@ protected function getPayload($key)
             return $this->emptyPayload();
         }
 
+        $e = null;
+
         try {
-            $data = unserialize(substr($contents, 10));
+            $data = @unserialize(substr($contents, 10));
```

Comment:
@taylorotwell @driesvints Previously, on situations where the `unserialize` was failing because of bad string content, the `$data` variable contained the value `false` - and no exception would be thrown. This is caused because the `unserialize` functions returns `false` and not an exception in this cases.

In PHPUnit 9, we are converting notices to exceptions, causing the `CacheFileStoreTest::testUnserializableFileContentGetDeleted` test to pass. 

Yet, in PHPUnit 10, there is no such as feature that converts notices to exceptions, effectively making our tests to behave exactly like user-land code would behave. Therefore, had to adjust this code - in the process probably fixing a bug - for when the result `$data` is `false`, we also forget the original `$key`.